From 4a7c524fc9b2c45451c871ceebf71c6407291ed4 Mon Sep 17 00:00:00 2001
From: Christophe Dumez <christophe.dumez@intel.com>
Date: Fri, 3 Jun 2011 09:05:46 +0300
Subject: [PATCH] QtVersit: Reverse lat/lon order for QContactGeolocation

Reverse the latitude/longiture order in the vCard representation
of QContactGeolocation to comply with RFC-2426 and improve
interoperability with other software such as libebook/EDS.

Upstream bug: QTMOBILITY-1589.
---
 src/versit/qversitcontactexporter_p.cpp |    5 +++--
 src/versit/qversitcontactimporter_p.cpp |    5 +++--
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/src/versit/qversitcontactexporter_p.cpp b/src/versit/qversitcontactexporter_p.cpp
index 8bdf997..8f992aa 100644
--- a/src/versit/qversitcontactexporter_p.cpp
+++ b/src/versit/qversitcontactexporter_p.cpp
@@ -487,8 +487,9 @@ void QVersitContactExporterPrivate::encodeGeoLocation(
     QContactGeoLocation geoLocation = static_cast<QContactGeoLocation>(detail);
     QVersitProperty property;
     property.setName(mPropertyMappings.value(detail.definitionName()));
-    property.setValue(QStringList() << QString::number(geoLocation.longitude())
-                      << QString::number(geoLocation.latitude()));
+    // RFC-2426 says the order should be "LAT LON"
+    property.setValue(QStringList() << QString::number(geoLocation.latitude())
+                      << QString::number(geoLocation.longitude()));
     property.setValueType(QVersitProperty::CompoundType);
     *generatedProperties << property;
     *processedFields << QContactGeoLocation::FieldLongitude
diff --git a/src/versit/qversitcontactimporter_p.cpp b/src/versit/qversitcontactimporter_p.cpp
index 040c6e3..88c9084 100644
--- a/src/versit/qversitcontactimporter_p.cpp
+++ b/src/versit/qversitcontactimporter_p.cpp
@@ -673,10 +673,11 @@ bool QVersitContactImporterPrivate::createGeoLocation(
             || variant.type() != QVariant::StringList)
         return false;
     QStringList values = variant.toStringList();
+    // RFC-2426 says the order should be "LAT LON"
     bool ok1;
-    geo.setLongitude(takeFirst(values).toDouble(&ok1));
+    geo.setLatitude(takeFirst(values).toDouble(&ok1));
     bool ok2;
-    geo.setLatitude(takeFirst(values).toDouble(&ok2));
+    geo.setLongitude(takeFirst(values).toDouble(&ok2));
 
     if (ok1 && ok2) {
         saveDetailWithContext(updatedDetails, geo, extractContexts(property));
-- 
1.7.4.4

